xds/src/main/java/io/grpc/xds/GrpcXdsTransportFactory.java

```diff
@@ -35,34 +36,45 @@
 final class GrpcXdsTransportFactory implements XdsTransportFactory {
 
   static final GrpcXdsTransportFactory DEFAULT_XDS_TRANSPORT_FACTORY =
-      new GrpcXdsTransportFactory();
+      new GrpcXdsTransportFactory(null);
+
+  private final CallCredentials callCredentials;
+
+  GrpcXdsTransportFactory(CallCredentials callCredentials) {
+    this.callCredentials = callCredentials;
+  }
 
   @Override
   public XdsTransport create(Bootstrapper.ServerInfo serverInfo) {
-    return new GrpcXdsTransport(serverInfo);
+    GrpcXdsTransport transport = new GrpcXdsTransport(serverInfo, callCredentials);
```

Comment:
nit: return directly without the local variable since you aren't calling set() any more?